json_object.c

```diff
@@ -720,22 +754,76 @@ int64_t json_object_get_int64(const struct json_object *jso)
 	}
 }
 
+uint64_t json_object_get_uint64(const struct json_object *jso)
+{
+	uint64_t cuint;
+
+	if (!jso)
+		return 0;
+	switch(jso->o_type)
+	{
+	case json_type_int:
+		switch(jso->o.c_int.cint_type) {
+		case json_object_int_type_int64:
+			if (jso->o.c_int.cint.c_int64 < 0)
+				return 0;
+			return (uint64_t)jso->o.c_int.cint.c_int64;
+		case json_object_int_type_uint64:
+			return jso->o.c_int.cint.c_uint64;
+		}
+	case json_type_double:
+		// UINT64_MAX can't be exactly represented as a double
+		// so cast to tell the compiler it's ok to round up.
+		if (jso->o.c_double >= (double)UINT64_MAX)
+			return UINT64_MAX;
+		if (jso->o.c_double < 0)
+			return 0;
+		return (uint64_t)jso->o.c_double;
+	case json_type_boolean:
+		return jso->o.c_boolean;
+	case json_type_string:
+		if (json_parse_uint64(get_string_component(jso), &cuint) == 0)
+			return cuint;
+		/* FALLTHRU */
+	default:
+		return 0;
+	}
+}
+
 int json_object_set_int64(struct json_object *jso,int64_t new_value){
-	if (!jso || jso->o_type!=json_type_int)
+	if (!jso || jso->o_type!=json_type_int || jso->o.c_int.cint_type != json_object_int_type_int64)
+		return 0;
+	jso->o.c_int.cint.c_int64=new_value;
+	return 1;
+}
+
+int json_object_set_uint64(struct json_object *jso,uint64_t new_value){
+	if (!jso || jso->o_type!=json_type_int || jso->o.c_int.cint_type != json_object_int_type_uint64)
 		return 0;
-	jso->o.c_int64=new_value;
+	jso->o.c_int.cint.c_uint64=new_value;
 	return 1;
 }
 
 int json_object_int_inc(struct json_object *jso, int64_t val) {
-	if (!jso || jso->o_type != json_type_int)
+	if (!jso || jso->o_type != json_type_int || jso->o.c_int.cint_type != json_object_int_type_int64)
```

Comment:
This shouldn't silently refuse to increment other cint_type.  Let's have it handle any cint_type, and if we happen to go past INT64_MAX, switch it to _type_uint64.

Reply:
ok, done. I have implemented the mutual conversion from int64 to uint64 in ```json_object_int_inc``` . I forgot this point before. It is necessary and shows superiorities of the new int data structure wihch you suggest.

Reply:
And, although you did not mark the ```json_object_uint_inc```, I have modified it as the same with ```json_object_int_inc```. 
> int64 +uint 64 ...